Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,47 @@
-function setAlarm() {}
+let alarmInterval;
+
+function setAlarm() {
+  const inputField = document.getElementById("alarmSet"); // Input field for seconds
+  
+  let inputValue = parseInt(inputField.value)//Get time in seconds
+
+  const timeDisplay = document.getElementById("timeRemaining"); // Time display element
+  
+  if (isNaN(inputValue) || inputValue <= 0){
+    alert ("Please enter a valid number of seconds");
+    return;
+  }
+  // Function to format time as MM:SS
+  function getFormattedTime(time) {
+    const minutes = Math.floor(time / 60).toString().padStart(2, "0");
+    const secs = (time % 60).toString().padStart(2, "0");
+    return `${minutes}:${secs}`;
+  }
+
+  timeDisplay.textContent = `Time Remaining: ${getFormattedTime(inputValue)}`
+
+  alarmInterval = setInterval (() => {
+    inputValue -= 1;
+
+    if (inputValue <= 0) {
+      clearInterval(alarmInterval);
+      timeDisplay.textContent = "Time remaining: 00:00";
+      playAlarm();
+      document.body.style.backgroundColor = "Pink";
+    }else {
+      timeDisplay.textContent = `Time remaining : ${getFormattedTime(inputValue)}`;
+    }
+    },1000);
+  }
+  function pauseAlarm (){
+    audio.pause();
+    clearInterval(countdownInterval);
+  }
```

Comment:
There was already a function named `pauseAlarm` in this file.
Which of these two functions will be used as the event listener for the "Stop Alarm" button?